pp_hot.c

```diff
@@ -3205,31 +3205,42 @@ PP(pp_match)
     /* push captures on stack */
 
     {
-        const I32 nparens = RXp_NPARENS(prog);
-        I32 i = (global && !nparens) ? 1 : 0;
+        const I32 logical_nparens = RXp_LOGICAL_NPARENS(prog);
+        I32 i = (global && !logical_nparens) ? 1 : 0;
+        I32 *l2p = RXp_LOGICAL_TO_PARNO(prog);
+        I32 *p2l_next = RXp_PARNO_TO_LOGICAL_NEXT(prog);
+        I32 paren, offs_start, offs_end;
```

Comment:
I expect the match start/end positions can be more than 2GB into the string.

Reply:
Yep